docs/strings.rst

```diff
@@ -72,10 +72,19 @@ In the file, these map to fixed-width ASCII strings.  One byte per character
 is used.  The representation is "null-padded", which is the internal
 representation used by NumPy (and the only one which round-trips through HDF5).
 
-Technically, these strings are supposed to store `only` ASCII-encoded text,
-although in practice anything you can store in NumPy will round-trip.  But
-for compatibility with other progams using HDF5 (IDL, MATLAB, etc.), you
-should use ASCII only.
+How these strings are read depends upon whether they are scalar attributes or
+not:
+
+- Scalar attributes are read as native Python strings. On Python 3, that means
+  decoding from ASCII to unicode. If decoding fails, the
```

Comment:
Thanks for noting this. This was actually intentional: I didn't want to give users ideas about storing data UTF-8 in HDF5 ASCII strings. But it's probably better to keep the keep the docs in sync with the actual implementation.